frontend/src/pages/main/topup/TopUp.tsx

```diff
@@ -1,11 +1,41 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import {Titles} from "../../../common/constansts/titles";
+import {Box} from "@mui/material";
+import Title from "../../../components/typography/title/Title";
+import TopUpConfirmation from "./confirm/TopUpConfirmation";
+import {IWallet} from "../../../types/wallet";
+import {useAppSelector} from "../../../stores/redux/hooks";
+import {selectLoggedUser} from "../../../stores/redux/loggedUser/loggedUserSlice";
+import {getWalletByUserService} from "../../../services/wallet";
+
+const initialWallet: IWallet = {
+    id: -1,
+    balance: 0,
+    userId: -1
+}
 
 const TopUp: React.FC = () => {
+
+    const [wallet, setWallet] = useState<IWallet>(initialWallet);
+    const loggedUser = useAppSelector(selectLoggedUser);
+
+    const getWallet = async () => {
+        const receivedWallet = await getWalletByUserService(loggedUser.id, loggedUser.userType);
```

Comment:
It's time to think about it. You get wallet id on every single page. Can you fetch it once, the reuse it everywhere?